crates/matrix-sdk-crypto/src/machine/tests/decryption_verification_state.rs

```diff
@@ -380,3 +381,253 @@ async fn test_verification_states_multiple_device() {
 
     assert_eq!(VerificationState::Unverified(VerificationLevel::UnverifiedIdentity), state);
 }
+
+#[async_test]
+async fn test_decryption_trust_requirement() {
+    let (alice, bob) = get_machine_pair_with_setup_sessions_test_helper(
+        tests::alice_id(),
+        tests::user_id(),
+        false,
+    )
+    .await;
+    let room_id = room_id!("!test:example.org");
+    let (event, session_id) = encrypt_message(&alice, room_id, &bob, "Secret message").await;
+
+    // Set the sender data to various values, and test that we can or can't
+    // decrypt, depending on what the trust requirement is.
```

Comment:
It's not obvious how this comment relates to the code immediately after it. Or does it apply to the whole test? it should maybe be a doc-comment on the function if so.

Reply:
In fact, the test needs a doc-comment describing what it is testing in any case.